ctapipe/reco/stereo_combination.py

```diff
@@ -0,0 +1,291 @@
+from abc import abstractmethod
+
+import astropy.units as u
+import numpy as np
+from astropy.table import Table
+
+from ctapipe.core import Component, Container
+from ctapipe.core.traits import Bool, CaselessStrEnum, Unicode
+
+from ..containers import (
+    ArrayEventContainer,
+    ParticleClassificationContainer,
+    ReconstructedEnergyContainer,
+)
+from .utils import add_defaults_and_meta
+
+_containers = {
+    "energy": ReconstructedEnergyContainer,
+    "classification": ParticleClassificationContainer,
+}
+
+__all__ = [
+    "StereoCombiner",
+    "StereoMeanCombiner",
+]
+
+
+def _calculate_ufunc_of_telescope_values(tel_data, n_array_events, indices, ufunc):
+    combined_values = np.zeros(n_array_events)
+    ufunc.at(combined_values, indices, tel_data)
+    return combined_values
+
+
+def _weighted_mean_ufunc(tel_values, weights, n_array_events, indices):
+    # avoid numerical problems by very large or small weights
+    weights = weights / weights.max()
+    sum_prediction = _calculate_ufunc_of_telescope_values(
+        tel_values * weights,
+        n_array_events,
+        indices,
+        np.add,
+    )
+    sum_of_weights = _calculate_ufunc_of_telescope_values(
+        weights, n_array_events, indices, np.add
+    )
+    mean = np.full(n_array_events, np.nan)
+    valid = sum_of_weights > 0
+    mean[valid] = sum_prediction[valid] / sum_of_weights[valid]
+    return mean
+
+
+class StereoCombiner(Component):
+    """Base Class for algorithms combining telescope-wise predictions to common prediction"""
+
+    prefix = Unicode(
+        default_value="",
+        help="Prefix to be added to the output container / column names",
+    ).tag(config=True)
+    property = CaselessStrEnum(
+        ["energy", "classification", "direction"],
+        help="Whic property is being combined",
+    ).tag(config=True)
+
+    @abstractmethod
+    def __call__(self, event: ArrayEventContainer) -> None:
+        """
+        Fill event container with stereo predictions
+        """
+
+    @abstractmethod
+    def predict_table(self, mono_predictions: Table) -> Table:
+        """
+        Constructs stereo predictions from a table of
+        telescope events.
+        """
+
+
+class StereoMeanCombiner(StereoCombiner):
+    """
+    Calculate array-event prediction as (weighted) mean of telescope-wise predictions
+    """
+
+    weights = CaselessStrEnum(
+        ["none", "intensity", "konrad"],
```

Comment:
Do we have a new name for konrad yet?

Reply:
No. It solve that globally in an another PR. Currently it's the same as in the geometry reconstructors.